portfolio/src/main/java/com/google/sps/servlets/CommentsServlet.java

```
@@ -34,28 +37,36 @@
 
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/comment-data")
-public class DataServlet extends HttpServlet {
+public class CommentsServlet extends HttpServlet {
+    UserService userService = UserServiceFactory.getUserService();
+    
     @Override
     public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
         response.setContentType("application/json");
         ArrayList<Comment> comments = new ArrayList<>();
         DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
         int lim = getNumComments(request);
-        int cnt = 0;
 
+        FetchOptions fetchOptions = FetchOptions.Builder.withLimit(lim);
         Query query = new Query("Comment").addSort("time", SortDirection.DESCENDING);
-        PreparedQuery results = datastore.prepare(query);
-        for (Entity entity : results.asIterable()) {
-            cnt++;
-            if(cnt > lim) break;
+        PreparedQuery pq = datastore.prepare(query);
+        QueryResultList<Entity> results;
+        try {
+            results = pq.asQueryResultList(fetchOptions);
+        } catch(Exception e) {
```

Only catch the exception that you want to handle. For example, the try block might throw java.lang.IllegalStateException. So you probably just want to catch this exception type. Otherwise, you might swallow some other exception types which will make debugging difficult if some other exception types occur. 

Also, think about the way to handle the exception. For example, the pq.asQueryResultList(fetchOptions) generates IllegalStateException. You can display an error message on the client or ask the users to refresh the webpage. Or you can call pq.asQueryResultList(fetchOptions) again if you think this is just a temperate issue.

---

Turns out I didn't need to handle the exception at all, so I just removed the try-catch.